src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -348,6 +349,11 @@ void Transformations::PreLpt(const std::vector<ov::element::Type>& defaultPrecis
 
     ov::pass::Manager manager;
     manager.set_per_pass_validation(false);
+
+    // Decomposition
+    auto decomp = manager.register_pass<ov::pass::GraphRewrite>();
+    ADD_MATCHER(decomp, ov::pass::MatmulGatherDecomposition)
```

Comment:
1. We don't need to register a separate `GraphRewrite` here.
2. Please use `CPU_REGISTER_PASS_COMMON` or `CPU_REGISTER_PASS_X64` (depending on whether the transformation is needed on ARM platforms)
3. Please register `CPU_REGISTER_PASS_COMMON(manager, ov::pass::Validate);` right after the pass (since the new pass changes several places in graph and it's better to validate the model after that)
4. Consider moving this pass lower at the end of pipeline: usually, at that stage graph is more optimal (e.g. constant subgraphs are folded if possible), and there is more possibility that the new transformation will be successfully matched

Reply:
Top 3 have been updated.
About 4: if I move this pass to the end of pipline, current pattern will mismatch, some special shape will bring different pattern, it will be more complex. 
Generally, merged matmul have better performance, but VIT model is special case, 3 gather have more compute, actually I just want to process VIT case, maybe we can put this pass to head. @v-Golubev